### PR TITLE
Made Section.add_student idempotent even when move_for_same_teacher i…

### PR DESCRIPTION
**Pull request by aoby**
…s false (e.g. for workshop sections). Added unit test

### CODE REVIEW

File: dashboard/app/models/section.rb [hunk 1]
```diff
@@ -74,7 +74,7 @@ def add_student(student, move_for_same_teacher: true)
       # new one
       follower.update_attributes!(section: self)
     else
-      follower = Follower.create!(user_id: self.user_id, student_user: student, section: self)
+      follower = Follower.find_or_create_by!(user_id: self.user_id, student_user: student, section: self)
```
Comment (breville) [inline dashboard/app/models/section.rb:77]: interesting that there's also a first_or_create, but this seems appropriate here.

File: dashboard/test/models/section_test.rb [hunk 2]
```diff
@@ -101,6 +101,20 @@ class SectionTest < ActiveSupport::TestCase
     assert new_section.students.exists?(student.id)
   end
 
+  test 'add_student is idempotent' do
```
Comment (breville) [inline dashboard/test/models/section_test.rb:104]: nice testing

**Comment by breville** [comment]
Was it possible to hit this issue with regular section join as well?

lgtm

**Comment by aoby** [comment]
The regular (non-workshop) section join did not hit this issue, because the `move_for_same_teacher` logic updated the existing entry and resulted in a no-op.

**Pull request merged**